### PR TITLE
Change LoggerObject to print stack trace

### LINKED ISSUE #108: Improvements to Log.error method
Currently we can only pass the exception to Log.error method. It does not accept any error message at the moment. It is better if we can improve this method as follows,

Log.error(e, message)

### PR DESCRIPTION
**Pull request by Shan1024**
Resolves #108 

This will also add a new private static class **CustomException** which will be used to modify and get the stack trace of the error.

### CODE REVIEW

File: components/uuf-renderablecreator-hbs/src/main/java/org/wso2/carbon/uuf/renderablecreator/hbs/impl/js/LoggerObject.java [hunk 1]
```diff
@@ -149,6 +168,28 @@ public static JsonElement serialize(Object obj) {
...
-                "function(obj)}";
+                "function(obj), error: function(message, obj)}";
+    }
+
+    /**
+     * Uses to get a modified stack trace.
+     *
+     * @since 1.0.0
+     */
+    private static class CustomException extends Exception {
```
Comment (Kishanthan) [inline components/uuf-renderablecreator-hbs/src/main/java/org/wso2/carbon/uuf/renderablecreator/hbs/impl/js/LoggerObject.java]: change the name of this class to be meaningful on what it does

Comment (this) [inline components/uuf-renderablecreator-hbs/src/main/java/org/wso2/carbon/uuf/renderablecreator/hbs/impl/js/LoggerObject.java]: `StackTracePrinter`

File: components/uuf-renderablecreator-hbs/src/main/java/org/wso2/carbon/uuf/renderablecreator/hbs/impl/js/LoggerObject.java [hunk 2]
```diff
@@ -74,6 +76,23 @@ public void error(Object obj) {
         logger.error(getLogMessage(obj));
     }
 
+    public void error(String message, Object obj) {
```
Comment (manuranga) [inline components/uuf-renderablecreator-hbs/src/main/java/org/wso2/carbon/uuf/renderablecreator/hbs/impl/js/LoggerObject.java]: @Shan1024 
what does this mean? can I do `Log.error("manu","ranga");` ?

Comment (Shan1024) [inline components/uuf-renderablecreator-hbs/src/main/java/org/wso2/carbon/uuf/renderablecreator/hbs/impl/js/LoggerObject.java]: @manuranga: Yes, you can. But as a best practice we advice that the second parameter to be used only to pass an error object. Then the stack trace will be logged along with the error and the custom message.

Comment (manuranga) [inline components/uuf-renderablecreator-hbs/src/main/java/org/wso2/carbon/uuf/renderablecreator/hbs/impl/js/LoggerObject.java]: If it's wrong why don't you throw an error? We can't teach all the people the best practice. and the variable name is obj? that's not nice.

**Pull request merged**